loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,111 @@
+
+    /**
+     * Defines whether this extension is enabled.
+     */
+    public enabled: boolean;
+
+    private _loader: GLTFLoader;
+
+    /** @hidden */
+    constructor(loader: GLTFLoader) {
+        this._loader = loader;
+        this.enabled = this._loader.isExtensionUsed(NAME);
+    }
+
+    /** @hidden */
+    public dispose() {
+        delete this._loader;
+    }
+
+    /** @hidden */
+    public loadNodeAsync(context: string, node: INode, assign: (babylonTransformNode: TransformNode) => void): Nullable<Promise<TransformNode>> {
+        return GLTFLoader.LoadExtensionAsync<IKHRMeshInstancing, TransformNode>(context, node, this.name, (extensionContext, extension) => {
+            const nodeMesh = node.mesh;
+            const promise = this._loader.loadNodeAsync(`#/nodes/${node.index}`, node, (babylonTransformNode) => {
+                babylonTransformNode.parent = this._loader.rootBabylonMesh;
+                const attributeBufferPromises = new Array<Promise<any>>();
+
+                let attributes = extension.attributes;
+                let attributeKeys = Object.keys(attributes);
+                const attributeBuffers : { [name: string]: { buffer : Float32Array, accessor : IAccessor } } = {};
+                for (let i = 0; i < attributeKeys.length; ++i) {
+                    let attributeKey = attributeKeys[i];
+                    let accessor = ArrayItem.Get(`${extensionContext}/attributes/${attributeKey}`, this._loader.gltf.accessors, attributes[attributeKey]);
+                    attributeBufferPromises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        attributeBuffers[attributeKey] = { buffer : data, accessor : accessor };
+                    }));
+                }
+
+                const mesh = ArrayItem.Get(`${context}/mesh`, this._loader.gltf.meshes, extension.mesh || nodeMesh);
+                this._loader._loadMeshAsync(`/meshes/${mesh.index}`, node, mesh, (babylonMeshInstanceNode) => {
+                    Promise.all(attributeBufferPromises).then(() => {
+                        attributeKeys = Object.keys(attributeBuffers);
+
+                        if (attributeKeys.length > 0) {
+                            let instanceCount = attributeBuffers[attributeKeys[0]].accessor.count;
+                            let digitLength = instanceCount.toString().length;
+                            for (let i = 0; i < instanceCount; ++i) {
+                                let instance;
+                                if (i === 0){
+                                    instance = babylonMeshInstanceNode;
+                                    instance.setParent(babylonTransformNode);
+                                } else {
+                                    instance = babylonMeshInstanceNode.clone((babylonMeshInstanceNode.name || babylonMeshInstanceNode.id) + "_" + String(i).padStart(digitLength,'0'), babylonTransformNode, true);
```

Comment:
`padStart` is not compatible with IE, so you can't use it :(

Reply:
Ah, my mistake. I created a scoped function to handle the padding.